### PR TITLE
Fix conversion of geometry db map keys

### LINKED ISSUE #929: [Code health] Harmonize geometry representation
Shapes are represented in remote db in features as `point` (GeoPoint), `geoJson` (string) and `geometry` (map). There's also some confusion in the implementation about geometries vs LOIs. This design wasn't intentional and we can improve it.

Ideas:
- [x] Define generic geometry types independent of Ground model (perhaps import ~[GeoTools](https://docs.geotools.org/latest/userguide/library/jts/geometry.html)?~ [JTS](https://github.com/locationtech/jts))
- [x] Instead of extending `LocationOfInterest`, have it contain an arbitrary `Geometry`
- [x] Replace separate point, geoJson db format mirror GeoJSON where appropriate, but use Firebase `GeoPoint` for coordinates. Avoid nested arrays since they're unsupported by Firebase.
- [x] Load/save in local db (@scolsen)
- [x] Update map to render new Geometry types (@jsunde)
- [x] Delete old subclasses of `LocationOfInterest` (@scolsen)

### PR DESCRIPTION
**Pull request by JSunde**
Fixes #929 as well as fixing a ClassCastException in GeometryConverter and returns Result from LoiConverter.

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/persistence/remote/firestore/schema/LoiConverter.kt [hunk 1]
```diff
@@ -34,18 +35,15 @@ object LoiConverter {
   const val GEOMETRY_COORDINATES = "coordinates"
   const val GEOMETRY = "geometry"
 
-  fun toLoi(survey: Survey, doc: DocumentSnapshot): LocationOfInterest {
+  fun toLoi(survey: Survey, doc: DocumentSnapshot): Result<LocationOfInterest> {
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/persistence/remote/firestore/schema/LoiConverter.kt:38]: You'll need to update the two callers to check the result and on failure log a debug warning that the LOI was skipped (id + message).

Comment (JSunde) [reply ground/src/main/java/com/google/android/ground/persistence/remote/firestore/schema/LoiConverter.kt]: I think that actually already happens. In the QuerySnapshotConverter I'm returning a Result.error() when the Result<LocationOfInterest> is a failure, and then in LocationOfInterestRepository#updateLocalLocationOfInterest() it skips and logs those invalid LOIs

Comment (JSunde) [reply ground/src/main/java/com/google/android/ground/persistence/remote/firestore/schema/LoiConverter.kt]: The Result<LocationOfInterest> ends up getting wrapped in a RemoteDataEvent<LocationOfInterest> in the QuerySnapshotConverter.

**Review by gino-m — APPROVED** [review]
:1st_place_medal: LGTM!

**Pull request merged**